pages/studentPortal/accountability/index.tsx

```diff
@@ -27,94 +28,91 @@ const accountabilityMenuData = {
-  { creationDate: "2024-11-20T12:00:00Z", isCompleted: true },
-  { creationDate: "2024-12-25T12:00:00Z", isCompleted: false },
-];
-
-const initialAccountabilityTaskState = (user) => ({
-  user_id: user.uid,
+// Initial State
+const getInitialAccountabilityTaskState = (
+  userId: string
+): AccountabilityTask => ({
+  user_id: userId,
   topic: "",
   subfield: "",
   description: "",
-  validator: "", // TODO: Auto-fill this later if needed
+  validator: "", // TODO: Slack Integration will cover this.
 });
 
-const AccountabilityDashboard = () => {
+const AccountabilityDashboard: React.FC = () => {
   const { user } = useAuth();
-  const [accountabilityTask, setAccountabilityTask] = useState(
-    initialAccountabilityTaskState(user)
-  );
+  const userId = user?.uid ?? "";
+
+  const [accountabilityTask, setAccountabilityTask] =
+    useState<AccountabilityTask>(getInitialAccountabilityTaskState(userId));
+
+  const { topic, subfield, description } = accountabilityTask;
 
-  const [insertAccountabilityTask, { loading, error }] = useMutation(
+  // Fetch user accountability data
+  const {
+    data,
+    loading: accountabilityLoading,
+    error: accountabilityError,
+  } = useQuery(GET_ACCOUNTABILITY_TASKS_BY_USER, {
+    variables: { userId },
+    skip: !userId,
+  });
+
+  // Mutation for inserting accountability task
+  const [insertAccountabilityTask, { loading: inserting }] = useMutation(
     INSERT_ACCOUNTABILITY_MUTATION
   );
 
-  const { user_id, topic, subfield, description } = accountabilityTask;
-
-  const handleChange = (name, value) => {
-    setAccountabilityTask((prevTask) => {
-      if (name === "topic") {
-        // If topic changes, reset subfield
-        return { ...prevTask, topic: value, subfield: "" };
-      } else {
-        // For other changes (including subfield), just update the specified field
-        return { ...prevTask, [name]: value };
-      }
-    });
+  // Handlers
+  const handleChange = (name: keyof AccountabilityTask, value: string) => {
+    setAccountabilityTask((prevTask) => ({
+      ...prevTask,
+      [name]: value,
+      ...(name === "topic" && { subfield: "" }), // Reset subfield if topic changes
+    }));
+  };
+
+  const validateForm = (): boolean => {
+    return Boolean(topic && subfield && description);
   };
 
-  // Handle form submission
   const handleSubmit = async () => {
-    const variables = { user_id, topic, subfield, description };
+    if (!validateForm()) {
+      alert("Please fill in all fields before submitting.");
+      return;
+    }
 
     try {
-      const { data } = await insertAccountabilityTask({ variables });
-      console.log("Task inserted:", data.insert_accountability_one);
-      resetTaskState(user, setAccountabilityTask);
+      const variables = { user_id: userId, topic, subfield, description };
+      const response = await insertAccountabilityTask({ variables });
+      console.log("Task inserted:", response.data.insert_accountability_one);
```

Comment:
Don't merge console logs into production. We don't want the public to see our raw data and ids. 

Usually harmless but if not careful could end up leaking private data.